From 40e1a01fe78feb099035c377848bce4fedc5c99c Mon Sep 17 00:00:00 2001
From: Ezra Buehler <ezra.buehler@husqvarnagroup.com>
Date: Sat, 17 Aug 2024 11:23:14 +0200
Subject: [PATCH] suricatta: Fix nocheckcert configuration

Without this change, setting `nocheckcert = true`, as found in
examples/configuration/swupdate.cfg, has no effect.

Signed-off-by: Ezra Buehler <ezra.buehler@husqvarnagroup.com>
(cherry picked from commit d720c15a380c2417f07b46a7526d1afda7007720)
Upstream-Status: Backport
---
 corelib/server_utils.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/corelib/server_utils.c b/corelib/server_utils.c
index 33c2d9a0..bed37efa 100644
--- a/corelib/server_utils.c
+++ b/corelib/server_utils.c
@@ -19,6 +19,7 @@
 int channel_settings(void *elem, void *data)
 {
 	char tmp[128];
+	bool tmp_bool;
 	channel_data_t *chan = (channel_data_t *)data;
 
 	GET_FIELD_INT(LIBCFG_PARSER, elem, "retry",
@@ -35,6 +36,8 @@ int channel_settings(void *elem, void *data)
 	if (strlen(tmp))
 		chan->retry_sleep =
 			(unsigned int)strtoul(tmp, NULL, 10);
+	GET_FIELD_BOOL(LIBCFG_PARSER, elem, "nocheckcert", &tmp_bool);
+	chan->strictssl = !tmp_bool;
 	GET_FIELD_STRING_RESET(LIBCFG_PARSER, elem, "cafile", tmp);
 	if (strlen(tmp))
 		SETSTRING(chan->cafile, tmp);
-- 
2.43.0

